### PR TITLE
Port BTC xpub to Rust

### PR DESCRIPTION
**Pull request by benma**
Builds on https://github.com/digitalbitbox/bitbox02-firmware/pull/684

### CODE REVIEW

File: src/rust/bitbox02-rust/src/hww/api/bitcoin.rs [hunk 1]
```diff
@@ -50,6 +57,47 @@ fn coin_name(coin: pb::BtcCoin) -> &'static str {
...
+) -> Result<Response, Error> {
+    let params = params::get(coin);
+    bitcoin::keypath::validate_xpub(keypath, params.bip44_coin)?;
+    let xpub_type = match xpub_type {
+        XPubType::Tpub => xpub_type_t::TPUB,
+        XPubType::Xpub => xpub_type_t::XPUB,
+        XPubType::Ypub => xpub_type_t::YPUB,
+        XPubType::Zpub => xpub_type_t::ZPUB,
+        XPubType::Vpub => xpub_type_t::VPUB,
+        XPubType::Upub => xpub_type_t::ZPUB,
```
Comment (x1ddos) [inline src/rust/bitbox02-rust/src/hww/api/bitcoin.rs:75]: typo. should be `xpub_type_t::UPUB`

File: src/rust/bitbox02-rust/src/hww/api/bitcoin.rs [hunk 2]
```diff
@@ -50,6 +57,47 @@ fn coin_name(coin: pb::BtcCoin) -> &'static str {
     }
 }
 
+/// Processes an xpub api call.
+async fn xpub(
```
Comment (x1ddos) [inline src/rust/bitbox02-rust/src/hww/api/bitcoin.rs:61]: Would be nice to have a simple test for each XPubType, valid input => valid output.

File: src/rust/bitbox02-rust/src/hww/api/bitcoin/params.rs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::pb;
+use pb::BtcCoin;
+
+use util::bip32::HARDENED;
+
+pub struct Params {
```
Comment (x1ddos) [inline src/rust/bitbox02-rust/src/hww/api/bitcoin/params.rs:20]: Would be nice to link to some docs like https://en.bitcoin.it/wiki/List_of_address_prefixes for example.

Also, what's the plan with btc_params.c? I assume we'll have only one at some point but they are duplicated now. I would cross reference this and btc_params.c in a comment.

**Review by x1ddos — APPROVED** [review]
LGTM but don't forget `xpub_type_t::UPUB`

File: src/rust/bitbox02-rust/src/hww/api/bitcoin.rs (hunk 1 above)
Comment (benma) [reply src/rust/bitbox02-rust/src/hww/api/bitcoin.rs:75]: omg

File: src/rust/bitbox02-rust/src/hww/api/bitcoin.rs (hunk 2 above)
Comment (benma) [reply src/rust/bitbox02-rust/src/hww/api/bitcoin.rs:61]: unfortunately the encoding happens in C, and we can only mock C calls in Rust unit tests.

**Pull request merged**